commit 51ba98ff6346007d252b2450ed4c1a479c41fb19
Author: Richard Hughes <richard@hughsie.com>
Date:   Thu Sep 29 14:32:17 2011 +0100

    power: Do not revert to the pre-idle brightness if idle dimming is disabled
    
    We want to set manager->priv->pre_dim_brightness = -1 to indicate that we're
    restored the value and that no further restoration is required, as the user may
    have already altered the level.
    
    This will only happen if you disable dimming at runtime after having dimmed at
    least once.
    
    Resolves https://bugzilla.gnome.org/show_bug.cgi?id=660434

diff --git a/plugins/power/gsd-power-manager.c b/plugins/power/gsd-power-manager.c
index f3c5ffe..5fbf42e 100644
--- a/plugins/power/gsd-power-manager.c
+++ b/plugins/power/gsd-power-manager.c
@@ -2733,6 +2733,7 @@ idle_set_mode (GsdPowerManager *manager, GsdPowerIdleMode mode)
                                 g_error_free (error);
                                 return;
                         }
+                        manager->priv->pre_dim_brightness = -1;
                 }
         }
 }
